code/_globalvars/misc.dm

```diff
@@ -20,3 +20,7 @@ var/map_ready = 0
 	However, using lots of spawn() has a severe performance impact, and often results in spaghetti-code
 	map_ready will be set to 1 when world/New() is called (which happens just after the map is instanced)
 */
+
+var/global/current_year = null
+var/global/current_month = null
+var/global/current_day = null

```

Comment:
Though since it is lazily initialised, you will have to add the 

```C++
current_year = text2num(time2text(world.realtime, "YYYY"))```

Every time you want to use it, otherwise you could have issues if it tries to access it before it was initialized.

Also since it uses the current time of the initialisation, it could cause issues if the current date is initialised, the server rolls over to the next day and then someone makes a new account.